packages/ui/src/post-maker/TweetGenAI/TweetGenAI.tsx

```diff
@@ -0,0 +1,158 @@
+    stop,
+    isLoading,
+    handleInputChange,
+    handleSubmit,
+  } = useCompletion({
+    api: 'api/route-tweet-gen',
+    body: {
+      numberOfPosts,
+      charLimit,
+      language,
+    },
+    onFinish(prompt, completion) {
+      setGeneratedPosts(JSON.parse(completion))
+    },
+    onError(error) {
+      toastMessage('Error', t('contact.form.failed'), 'error')
+    },
+  })
+
+  return (
+    <>
+      <Heading p={4} color="black">
+        Post Generator
+      </Heading>
+      <Box p={4}>
+        <form onSubmit={handleSubmit}>
+          <FormControl>
+            <FormLabel>Content</FormLabel>
+            <Textarea
+              name="prompt"
+              placeholder="Enter a content..."
+              value={input}
+              onChange={handleInputChange}
+              mb={4}
+              size="xs"
+              required
+            />
+          </FormControl>
+          <Flex
+            gap={[4, 4, 10]}
+            mb={4}
+            flexDirection={['column', 'column', 'row']}
+          >
+            <FormControl>
+              <FormLabel>Number of Posts</FormLabel>
+              <NumberInput
+                step={1}
+                min={0}
+                max={40}
+                defaultValue={5}
+                onChange={(a, b) => setNumberOfPosts(b)}
+              >
+                <NumberInputField />
+                <NumberInputStepper>
+                  <NumberIncrementStepper />
+                  <NumberDecrementStepper />
+                </NumberInputStepper>
+              </NumberInput>
+            </FormControl>
+            <FormControl>
+              <FormLabel>Character Limit</FormLabel>
+              <NumberInput
+                step={10}
+                min={10}
+                max={200}
+                defaultValue={150}
+                onChange={(a, b) => setCharLimit(b)}
+              >
+                <NumberInputField />
+                <NumberInputStepper>
+                  <NumberIncrementStepper />
+                  <NumberDecrementStepper />
+                </NumberInputStepper>
+              </NumberInput>
+            </FormControl>
+            <FormControl>
+              <FormLabel>Language</FormLabel>
+              <Select
+                value={language ?? ''}
+                onChange={e => setLanguage(e.target.value)}
+              >
+                {LANGUAGE_OPTIONS.map((opt, idx) => {
+                  return <option key={idx}>{opt}</option>
+                })}
+              </Select>
+            </FormControl>
+          </Flex>
+
+          <Button disabled={isLoading} type="submit">
+            {t('generate')}
+          </Button>
+          <Button type="button" onClick={stop} ml={2} colorScheme="gray">
+            Stop
+          </Button>
+        </form>
+      </Box>
+      {isLoading ? (
+        <Box p={4}>
+          <Progress size="xs" isIndeterminate />
+          {/* <div>{completion}</div> */}
+        </Box>
```

Comment:
Shall we display the completion for an effect? :)

```suggestion
        <Stack p={4}>
          <Progress size="xs" isIndeterminate />
          <Box color={'gray.500'}>{completion}</Box>
        </Box>
```

Reply:
I'll replace completion for the progress bar then. 😎